### PR TITLE
1235 intensive and extensive projections

### PR DESCRIPTION
**Pull request by IngridKJ**
## Proposed changes

Addresses point 1. and 2. of #1235. Specifically, this PR covers a tutorial for explaining/demonstrating projections of intensive and extensive quantities. This is done both in a more "theoretical"/conceptual way, and by code examples.

Additionally this PR addresses an issue with boundary grids after creating a non-matching mixed-dimensional grid. The examples in the tutorial are performed on a grid with one fracture, where the fracture grid has been refined by a factor of two, making it not match the rock matrix grid. After this refinement had been done, there was a need of calling compute_geometry for the boundary grids.

Credit to @mikeljordan for the first tutorial draft.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [x] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [x] Other: Tutorial

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [ ] The documentation is up-to-date.
- [ ] Static typing is included in the update.
- [ ] This PR does not duplicate existing functionality.
- [ ] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: tutorials/issue1235.ipynb [hunk 1]
```diff
@@ -0,0 +1,494 @@
+{
```
Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <p>We need an explanation of the concepts of intensive and extensive quantities. Also, I think the list should be moved later, e.g. to the subsections "Projection of intensive/extensive quantities".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <p>Wouldn't it be cleaner to override set_geometry?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <p>*visualizes</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <p>However, as shown in the second plot, the fracture itself has four cells. This means that...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <p>Could we have a slightly better definition of what we want to do? I.e., spell out what it means to project an intensive quantity. I suggest dedicating the first paragraph to this before proceeding to "The projections are done..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <p>I don't like the names. To me, "to_lower" sounds like a name of the projection, not the projected quantity.</p><p>Also, could you please add a comment on the use of trace? I.e. explain that we need a face quantity in the higher-dimensional subdomain?</p><p>Btw., could you please check whether we use "higher dimensional" or "higher-dimensional" elsewhere? I suspect it's the latter.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <p>This is a bit confusing. Let's talk through it and see if we can improve.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <p>I like this figure and explanation. Please indicate what's a fracture and what is an interface, though.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <p>Again, I take objection with the use of from and to.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (IvarStefansson) [inline tutorials/issue1235.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #11.    print(flux_to_l_value)
</pre><p>Perhaps add a "Fluxes projected using MortarProjection" to the print?</p><p>And "Fluxes projected by hard-coded indices" below</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (IngridKJ) [reply tutorials/issue1235.ipynb:1]: Regarding hyphen/no hyphen in "higher-dimensional": The majority of the occurrences throughout the codebase includes the hyphen. There are however a significant amount of non-hyphenated ones (1/3). For "lower-dimensional" vs. "lower dimensional": About a sixth of the occurrences do not include hyphen.

**Comment by IvarStefansson** [comment]
Please rename image files. Otherwise, I'm happy when the tests pass. Thanks for hunting down some bugs during the work on this PR!

File: tutorials/subdomain_and_interface_projections.ipynb [hunk 2]
```diff
@@ -0,0 +1,525 @@
+{
```
Comment (keileg) [inline tutorials/subdomain_and_interface_projections.ipynb:1]: <p>First sentence: I would make the (obvious when you know it) remark that the non-matching grid is needed to see the difference between intensive and extensive quantities. Or a different comment to that effect.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (keileg) [inline tutorials/subdomain_and_interface_projections.ipynb:1]: <p>Suggested rewrite, to make this appear a bit less like black magic:</p><p>Specifically, we have the following grid-relations (not visible from the plots, but available through inspection) which we will use later:</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (keileg) [inline tutorials/subdomain_and_interface_projections.ipynb:1]: <p>I like the explanation, and the figure is really nice. However, the overall text makes it appear that this weighting is only relevant for intensive quantities, and the reader will need to do some interpretation to understand what is different and what is similar for intensive quantities. Ideally, I would have liked to see this explanation given before the pressure example above, with an additional illustration of the projection of intensive quantities.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1286/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Comment by keileg** [comment]
@IngridKJ @IvarStefansson: Some comments to the notebook. The last one is somewhat substantial.

**Review by keileg — APPROVED** [review]
Nice work. Thanks also for the stamina, @IngridKJ, despite running into both https://github.com/pmgbergen/porepy/pull/1280 and https://github.com/pmgbergen/porepy/pull/1268

**Pull request merged**